app/controllers/bridges/bird/tasks_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Bridges
+  module Bird
+    class TasksController < ActionController::API
+      def index
+        builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') { |xml| build_bird(xml) }
+        render xml: builder
+      end
+
+      private
+
+      def build_bird(xml)
+        xml.import(
+          xmlns: 'https://www.mathplan.de/moses/xsd/default',
+          'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance'
+        ) do
+          bird_academy(xml)
+          bird_courses(xml)
+        end
+      end
+
+      def bird_academy(xml)
+        xml.list(type: 'bird_academy') do
+          xml.bird_academy do
+            xml.id_ codeharbor_id
+            xml.name 'CodeHarbor: Your repository for auto-gradeable programming exercises'
+            xml.kurzname 'CodeHarbor'
+          end
+        end
+      end
+
+      def bird_courses(xml)
+        xml.list(type: 'bird_course') do
+          tasks.each.with_index do |task, i|
+            xml.bird_course do
+              xml.id_ task_url(id: "sample-#{i + 1}") # TODO: This needs to be replaced with the actual task id.
+              xml.name task.title
+              xml.academyId codeharbor_id
+              xml.lectureType 'Programming Exercise'
+            end
+          end
+        end
+      end
+
+      def codeharbor_id
+        root_url
+      end
+
+      def tasks
+        # TODO: Replace with DB query.
```

Comment:
Potentially yes. I added a few demo exercises (for my screenshots) before importing all exercises from CodeOcean (still on the list when the proforma gem is updated to the newest version and all models related to the tasks are fixed).

Here's the current list of tasks:

<img width="1477" alt="Bildschirmfoto 2021-12-14 um 10 10 02" src="https://user-images.githubusercontent.com/7300329/145968226-9d282ac1-d021-46fe-b357-e63d9824d7d3.png">

Reply:
Great, thanks. I still would like to keep the dummy data response, as BIRD has no pagination mechanism so far and this is an index route.

Reply:
Fine for me. Let's merge and deploy it and ask the BIRD team to verify it.